### PR TITLE
Adding the ability to choose MC trails for training and testing

### PR DESCRIPTION
**Pull request by mohazahran**

### CODE REVIEW

File: python/ml4ir/applications/ranking/model/layers/masking.py [hunk 1]
```diff
@@ -160,8 +160,6 @@ def __init__(self,
         self.mask_at_inference = mask_at_inference
         self.requires_mask = requires_mask
 
-        super(QueryFeatureMask, self).__init__(**kwargs)
```
Comment (lastmansleeping) [inline python/ml4ir/applications/ranking/model/layers/masking.py:163]: @mohazahran you should still do `super().__init__(**kwargs)

File: python/ml4ir/base/config/keys.py [hunk 2]
```diff
@@ -135,7 +135,8 @@ class CalibrationKey(Key):
 
 class MonteCarloInferenceKey(Key):
     MONTE_CARLO_INFERENCE_TRIALS = "monte_carlo_inference_trials"
-    NUM_TRIALS = "num_trials"
+    NUM_TEST_TRIALS = "num_test_trials"
```
Comment (lastmansleeping) [inline python/ml4ir/base/config/keys.py:138]: @mohazahran Maybe `num_trials` and `num_trials_inference`?

And then if `num_trials_inference` is missing, you can use `num_trials`?

File: python/ml4ir/base/model/scoring/scorer_factory.py [hunk 3]
```diff
@@ -67,8 +67,9 @@ def get_scorer(model_config: dict,
     """
 
     if (MonteCarloInferenceKey.MONTE_CARLO_INFERENCE_TRIALS in model_config and
-            model_config[MonteCarloInferenceKey.MONTE_CARLO_INFERENCE_TRIALS].get(MonteCarloInferenceKey.NUM_TRIALS,
-                                                                                  0)):
+            (model_config[MonteCarloInferenceKey.MONTE_CARLO_INFERENCE_TRIALS].get(MonteCarloInferenceKey.NUM_TEST_TRIALS, 0) or
+            model_config[MonteCarloInferenceKey.MONTE_CARLO_INFERENCE_TRIALS].get(MonteCarloInferenceKey.NUM_TRAINING_TRIALS, 0))):
```
Comment (lastmansleeping) [inline python/ml4ir/base/model/scoring/scorer_factory.py:71]: @mohazahran if you use just `num_trials`, it'll simplify all this.

File: python/ml4ir/base/tests/test_monte_carlo_scorer.py [hunk 4]
```diff
@@ -12,7 +12,7 @@ def setUp(self):
         model_config = {'architecture_key': 'linear',
                         'layers': [{'type': 'dense', 'name': 'linear_layer', 'units': 1, 'activation': None}],
                         'optimizer': {'key': 'adam'}, 'lr_schedule': {'key': 'constant', 'learning_rate': 0.01},
-                        'monte_carlo_inference_trials': {'num_trials': 10}}
+                        'monte_carlo_inference_trials': {'num_test_trials': 10, "num_training_trials": 10}}
```
Comment (lastmansleeping) [inline python/ml4ir/base/tests/test_monte_carlo_scorer.py:15]: @mohazahran Add unit test for both variables

File: python/ml4ir/base/config/keys.py [hunk 5]
```diff
@@ -135,7 +135,8 @@ class CalibrationKey(Key):
 
 class MonteCarloInferenceKey(Key):
     MONTE_CARLO_INFERENCE_TRIALS = "monte_carlo_inference_trials"
```
Comment (lastmansleeping) [inline python/ml4ir/base/config/keys.py:137]: @mohazahran you can change this to `MONTE_CARLO_TRIALS` as it is not limited to `inference`

File: python/ml4ir/base/config/keys.py (hunk 2 above)
Comment (mohazahran) [reply python/ml4ir/base/config/keys.py:138]: @lastmansleeping I think being explicit of configuring the MC trails count for test and train is more clear and avoid confusion IMO

Comment (lastmansleeping) [reply python/ml4ir/base/config/keys.py:138]: [Nit] Okay, can you change to `num_trials_training` and `num_trials_inference`?

File: python/ml4ir/applications/ranking/model/layers/masking.py (hunk 1 above)
Comment (lastmansleeping) [reply python/ml4ir/applications/ranking/model/layers/masking.py:163]: ping

**Review by lastmansleeping — CHANGES_REQUESTED** [review]
@mohazahran Please address/reply to all comments before posting for re-review.

File: python/ml4ir/applications/ranking/model/layers/masking.py (hunk 1 above)
Comment (mohazahran) [reply python/ml4ir/applications/ranking/model/layers/masking.py:163]: @lastmansleeping It's being called in a previous line. This was duplicate

File: python/ml4ir/base/model/scoring/scorer_factory.py (hunk 3 above)
Comment (mohazahran) [reply python/ml4ir/base/model/scoring/scorer_factory.py]: same as my previous comment. I prefer being explicit.

**Review by lastmansleeping — APPROVED** [review]
